server/neptune/temporal/client.go

```diff
@@ -1,26 +1,60 @@
 package temporal
 
 import (
+	"context"
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"io"
+	"time"
 
 	"github.com/runatlantis/atlantis/server/core/config/valid"
 	"github.com/uber-go/tally/v4"
 	"go.temporal.io/sdk/client"
 	temporaltally "go.temporal.io/sdk/contrib/tally"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/interceptor"
 	"go.temporal.io/sdk/workflow"
 	"logur.dev/logur"
 )
 
-func NewClient(scope tally.Scope, logger logur.Logger, cfg valid.Temporal) (client.Client, error) {
+const StatsNamespace = "temporalclient"
+
+type Options struct {
+	Interceptors  []interceptor.ClientInterceptor
+	StatsReporter tally.StatsReporter
+}
+
+func (o *Options) WithClientInterceptors(i ...interceptor.ClientInterceptor) *Options {
+	o.Interceptors = i
+	return o
+}
+
+func (o *Options) WithStatsReporter(reporter tally.StatsReporter) *Options {
+	o.StatsReporter = reporter
+	return o
+}
+
+func NewClient(logger logur.Logger, cfg valid.Temporal, options *Options) (client.Client, error) {
 	opts := client.Options{
 		Namespace:          cfg.Namespace,
-		MetricsHandler:     temporaltally.NewMetricsHandler(scope),
 		Logger:             logur.LoggerToKV(logger),
 		ContextPropagators: []workflow.ContextPropagator{&ctxPropagator{}},
+		Interceptors:       options.Interceptors,
+	}
+
+	// default to app scope unless specified otherwise
+	var clientScope tally.Scope
+	var clientScopeCloser io.Closer
+	if options.StatsReporter != nil {
+		clientScope, clientScopeCloser = tally.NewRootScope(tally.ScopeOptions{
+			Prefix:   StatsNamespace,
+			Reporter: options.StatsReporter,
+		}, time.Second)
```

Comment:
do we want to log/error if stats reporter was nil?

Reply:
nah im kind of thinking of this package now as a standalone library that we might let others use.  Seems fair to have metrics as optional if for example a service doesn't use tally.